File: feign/simple-example/src/main/java/br/com/leonardoferreira/poc/feign/client/HttpBinClient.java
package br.com.leonardoferreira.poc.feign.client;

import br.com.leonardoferreira.poc.feign.client.config.HttpBinClientConfig;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;

@FeignClient(
        name = "httpBinClient",
        url = "${httpbin.url}",
        configuration = {
                HttpBinClientConfig.class
        }
)
public interface HttpBinClient {

    @GetMapping("/status/500")
    String retry();

}

File: multi-module-gradle-spring-boot/client-library/src/main/kotlin/br/com/leonardoferreira/poc/client/HttpBinClient.kt
package br.com.leonardoferreira.poc.client

import br.com.leonardoferreira.poc.UUIDResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping

@FeignClient(name = "httpBinClient", url = "\${feign.client.config.httpBinClient.url}")
interface HttpBinClient {

    @GetMapping("/uuid")
    fun findUUID(): UUIDResponse

} 
